src/DynamoDbQueryBuilder.php

```diff
@@ -315,15 +325,23 @@ protected function conditionsContainKey()
         return $conditionValue;
     }
 
+    /**
+     * Check if conditions "where" contain index key
+     * For composite index key, it will return false if the conditions don't have all composite key.
+     *
+     * @return array|bool false or array of keys
+     */
     protected function conditionsContainIndexKey()
     {
         if (empty($this->where)) {
             return false;
         }
 
         foreach ($this->model->getDynamoDbIndexKeys() as $key => $name) {
-            if (isset($this->where[$key])) {
-                return $key;
+            $keys          = $this->explodedIndexKeys($key);
```

Comment:
Ah, there's a problem.
If we declare like this
```php
    protected $dynamoDbIndexKeys = [
        'simple_index_name' => [
           'hash' => 'index_hash_key',
        ],
        'composite_index_name' => [
           'hash' => 'index_hash_key',
           'range' => 'index_range_key',
        ],
    ];
```
It will break backward compatibility with current declaration because index key and name are switched.
```php
    protected $dynamoDbIndexKeys = [
        'index_key' => 'index_name'
    ];
```
What do you think?